### DISCUSSION TITLE
ISR does not work after deploying next.js to vercel.

### DISCUSSION THREAD

**Discussion by ehdrjs4502** [vercel/next.js#54444] [Help]
### Summary

Logs when building.

Route (pages)                                Size     First Load JS
--
22:04:09.390 | ┌ ● / (ISR: 10 Seconds) (1812 ms)            4.83 kB         234 kB
22:04:09.390 | ├   /_app                                    0 B             123 kB
22:04:09.390 | ├ ○ /404                                     184 B           123 kB
22:04:09.391 | ├ ○ /category                                873 B           238 kB
22:04:09.391 | ├ ○ /category/[...cid]                       916 B           238 kB
22:04:09.391 | ├ ○ /guestbook                               1.45 kB         235 kB
22:04:09.391 | ├ ○ /login                                   3.51 kB         174 kB
22:04:09.391 | ├ ● /post/[pid] (ISR: 10 Seconds) (7022 ms)  287 kB          516 kB
22:04:09.391 | ├   ├ /post/27 (1165 ms)
22:04:09.391 | ├   ├ /post/28 (647 ms)
22:04:09.391 | ├   ├ /post/26 (591 ms)
22:04:09.391 | ├   ├ /post/23 (589 ms)
22:04:09.391 | ├   ├ /post/25 (588 ms)
22:04:09.391 | ├   ├ /post/6 (587 ms)
22:04:09.391 | ├   ├ /post/22 (586 ms)
22:04:09.391 | ├   └ [+3 more paths] (avg 579 ms)
22:04:09.391 | ├ ○ /post/edit                               1.93 kB         223 kB
22:04:09.391 | ├   └ css/4dfb736971b670d1.css               2.81 kB
22:04:09.391 | ├ ○ /search/[sid]                            1.17 kB         239 kB
22:04:09.391 | ├ ○ /signup                                  2.16 kB         172 kB
22:04:09.391 | ├ ● /tag (ISR: 10 Seconds) (606 ms)          1.94 kB         223 kB
22:04:09.391 | ├ ○ /tag/[tid]                               910 B           238 kB
22:04:09.391 | └ ○ /userinfo                                11.4 kB         185 kB
22:04:09.391 | + First Load JS shared by all                123 kB
22:04:09.391 | ├ chunks/framework-f76efef75eeb704a.js     45.2 kB
22:04:09.391 | ├ chunks/main-75d097dcb5321d78.js          32.3 kB
22:04:09.391 | ├ chunks/pages/_app-90c813cb3570ff60.js    43.7 kB
22:04:09.391 | ├ chunks/webpack-ca89a9e6313d59b9.js       1.85 kB
22:04:09.391 | └ css/1d3198177af16532.css                 141 B
22:04:09.391 |  
22:04:09.391 | ○  (Static)  automatically rendered as static HTML (uses no initial props)
22:04:09.392 | ●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
22:04:09.392 | (ISR)     incremental static regeneration (uses revalidate in getStaticProps)

![image](https://github.com/vercel/next.js/assets/33916924/02064ccc-9d9a-41e6-976c-2c79f88a7b7a)

If the data changes, it should be updated, but it will not.

The following error occurs in the execution log.

![image](https://github.com/vercel/next.js/assets/33916924/935af94d-7f7c-4314-a68b-f1db0b548df1)

How can I fix this error? I'd appreciate your help.

### Additional information

```js
export async function getStaticProps() {
    const API_URL = process.env.API

    // //최신 게시글 목록 가져오기 (최대 6개 가져옴)
    const latestPostRes: any = await axios.get(`${API_URL}lists/posts?pageSize=6`, {
        headers: {
            'ngrok-skip-browser-warning': '1234',
        },
    })

    const latestPostList = latestPostRes.data.postList || null


    return {
        props: { latestPostList },
        revalidate: 10,
    }
}
```

**Comment by hyjoong** [comment]
The reason for the error is not due to using ISR (Incremental Static Regeneration) but because the data fetching failed, and an attempt was made to return data that might not exist. For instance, in the code snippet below, latestPostRes might not have a value, yet an attempt to access latestPostRes.data.postList is made, resulting in an error.

To resolve the error, you have two options:

1. Use optional chaining to handle potential null or undefined values. This way, if latestPostList doesn't exist, undefined will be assigned.
```javascript
const latestPostList = latestPostRes?.data?.postList
```

2. Use a try-catch block to handle exceptions when data fetching fails. By doing this, errors won't propagate, and you can provide appropriate error handling.
```javascript
try {
  const latestPostRes: any = await axios.get(`${API_URL}lists/posts?pageSize=6`, {
        headers: {
            'ngrok-skip-browser-warning': '1234',
        },
    })

   const latestPostList = latestPostRes.data.postList
  return {
    props: { latestPostList },
    revalidate: 10,
  ]
} catch (error) {
  return {
    props: { /* Handle the error */ },
  }
}
```

These two approaches provide solutions for gracefully handling errors and potential data fetching issues. However, using a try-catch block seems more preferable as it not only prevents errors from propagating but also allows for explicit error handling.
